include/bluetooth/hci_raw.h

```diff
@@ -32,7 +32,86 @@ extern "C" {
  */
 int bt_send(struct net_buf *buf);
 
-/** @brief Enable Bluetooth RAW channel
+enum {
+	/** Passthrough mode
+	 *
+	 *  While in this mode the buffers are passed as is between the stack
+	 *  and the driver.
+	 */
+	BT_RAW_MODE_PASSTHROUGH = 0x00,
+
+	/** H:4 mode
+	 *
+	 *  While in this mode H:4 headers will added into the buffers
+	 *  according to the buffer type when comming from the stack and will be
+	 *  removed and used to set the buffer type.
+	 */
+	BT_RAW_MODE_H4 = 0x01,
+};
+
+/** @brief Set Bluetooth RAW channel mode
+ *
+ *  Set access mode of Bluetooth RAW channel.
+ *
+ *  @param mode Access mode.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_raw_set_mode(u8_t mode);
+
+/** @brief Get Bluetooth RAW channel mode
+ *
+ *  Get access mode of Bluetooth RAW channel.
+ *
+ *  @return Access mode.
+ */
+u8_t bt_raw_get_mode(void);
+
+#define BT_HCI_ERR_EXT_HANDLED  0xff
+
+/** Helper macro to define a command extention
+ *
+ *  @param _op Opcode of the command.
+ *  @param _min_len Minimal length of the command.
+ *  @param _func Handler function to be called.
+ */
+#define BT_RAW_CMD_EXT(_op, _min_len, _func) \
+	{ \
+		.op = _op, \
+		.min_len = _min_len, \
+		.func = _func, \
+	}
+
+struct bt_raw_cmd_ext {
+	/** Opcode of the command */
+	u16_t  op;
+
+	/** Minimal length of the command */
+	size_t min_len;
+
+	/** Handler function.
+	 *
+	 *  Handler function to be called when a command is intercepted.
+	 *
+	 *  @param buf Buffer containing the command.
+	 *
+	 *  @return HCI Status code or BT_HCI_ERR_EXT_HANDLED if command has
```

Comment:
Maybe add note about returning `0` to send the command for  handling in the controller?

Reply:
Done.